django/db/models/sql/compiler.py

```diff
@@ -795,6 +795,9 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     result += distinct_result
                     params += distinct_params
 
+                if self.query.comments:
```

Comment:
Let's put the coment right after SELECT, before the DISTINCT clause which can be arbitrarily long.

Reply:
Done, added a test as well.